rfd/0073-public-image-registry.md

```diff
@@ -0,0 +1,254 @@
+---
+authors: Logan Davis (logan.davis@goteleport.com)
+state: draft
+---
+
+# RFD 73 - Public Image Registry
+
+## What
+Teleport images are currently hosted on [Quay.io](https://quay.io/organization/gravitational). This RFD proposes migrating public images from Quay to [Amazon ECR](https://aws.amazon.com/ecr/). 
+
+## Why
+As of August 1st, 2021 Quay.io no longer supports any other authentication provider other than Red Hat Single-Sign On. Users in the Gravitational organization on Quay must be manually removed when they leave Teleport which presents a potential security risk. Migrating to Amazon ECR will consolidate our infrastructure while also improving security with support for IAM policies and our existing SSO infrastructure.
+
+## Details
+Teleport will use Amazon ECR and Amazon ECR Public in order to host public container images. The multiple stage registry pipeline will allow Teleport to test and verify images internally before promoting them to our customers. As of authoring this RFD, Amazon ECR Public lacks support for [vulnerability scanning and tag immutabilty](https://github.com/aws/containers-roadmap/issues/1288). The two-stage pipeline will allow us to leverage these features in the internal repository before pushing to the public. 
```

Comment:
What about GCP? We're already using it for some of our images and GCP offers images scanning (and also lacks of tag immutability AFAIK)

Reply:
[GCR](https://cloud.google.com/container-registry) and [Artifact Registry](https://cloud.google.com/artifact-registry) are both decent solutions for a container registry. From what I understand, they support vulnerability scanning with a cost associated, albeit small ($0.26 per image scan), whereas AWS ECR has a free tier. 

There doesn't seem to be a specific feature difference between the two. We already host cloud images on ECR as well as our other artifacts such as terraform provider and linux packages. Is there any specific reason we should leverage GCP over AWS in this case?